bears/vcs/VCSCommitBear.py

```diff
@@ -0,0 +1,131 @@
+import abc
+import re
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.HiddenResult import HiddenResult
+from coalib.misc.Shell import run_shell_command
+from collections import namedtuple
+from coala_utils.decorators import (enforce_signature, generate_ordering,
+                                    generate_repr)
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'raw_commit_message',
+               'commit_sha',
+               'commit_type',
+               'modified_files',
+               'added_files',
+               'deleted_files',
+               'message')
+@generate_ordering('raw_commit_message',
+                   'commit_sha',
+                   'commit_type',
+                   'modified_files',
+                   'added_files',
+                   'deleted_files',
+                   'severity',
+                   'confidence',
+                   'origin',
+                   'message_base',
+                   'message_arguments',
+                   'aspect',
+                   'additional_info',
+                   'debug_msg')
+class CommitResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin, raw_commit_message: str,
+                 commit_sha: str, commit_type: list, modified_files: list,
+                 added_files: list, deleted_files: list):
+        HiddenResult.__init__(self, origin, '')
```

Comment:
Can you help me understand why did you split all of these, why a named_tuple being passed here wasn't sufficient and why are you setting the origin of the result to be `CommitResult` class rather than the origin parameter ?